main.go

```diff
@@ -387,3 +418,68 @@ func humanDuration(duration time.Duration) string {
 
 	return v
 }
+
+func parseInclude(include string) []string {
+	repos := strings.Split(include, ",")
+	for i, repo := range repos {
+		repos[i] = strings.TrimSpace(repo)
+	}
+	return repos
+}
+
+func parseIncludeFromFile(filepath string) ([]string, error) {
+	if filepath == "" {
+		return nil, nil
+	}
+
+	reposBytes, err := os.ReadFile(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	lines := strings.Split(string(reposBytes), "\n")
+	repos := make([]string, 0, len(lines))
+	for _, line := range lines {
+		repo := strings.TrimSpace(line)
+
+		if repo != "" {
+			repos = append(repos, repo)
+		}
+	}
+
+	return repos, nil
+}
+
+// getFilter creates a filter map from a comma-separated list of repository names in the
+// format 'user/repo'
+func getFilterMap(repos []string) (map[string]bool, error) {
+	filter := make(map[string]bool)
+
+	for _, repo := range repos {
+		if !strings.Contains(repo, "/") {
+			return nil, fmt.Errorf("invalid repository name format: %q, must be in the format 'user/repo'", repo)
+		}
+		filter[repo] = true
+	}
+
+	return filter, nil
+}
+
+// reduceRepositories takes a list of repositories and a filter map, and returns a filtered list
+// based on the repository names.
+func reduceRepositories(allRepos []*github.Repository, filter map[string]bool) ([]*github.Repository, error) {
+	var filteredRepos []*github.Repository
+
+	for _, repo := range allRepos {
+		name := repo.GetFullName()
+		if filter[name] {
+			filteredRepos = append(filteredRepos, repo)
+		}
+	}
+
+	if len(filteredRepos) == 0 {
+		return nil, errors.New("no matching repositories found based on the provided filter")
```

Comment:
Why use errors.New here and fmt.Errorf above? On the whole in our codebases we tend to use fmt.Errorf

Reply:
I use `fmt.Errorf` when I need to provide dynamic data, i.e `repo`. I used `errors.New` in here because I wasn't injecting any dynamic data into the error string. 
I'd modify this to follow convention in my next update. Thank you